hamilton/function_modifiers/metadata.py

```diff
@@ -166,3 +167,94 @@ def decorate_node(self, node_: node.Node) -> node.Node:
         new_tags = node_.tags.copy()
         new_tags.update(self.tag_mapping.get(node_.name, {}))
         return tag(**new_tags).decorate_node(node_)
+
+
+SchemaTypes = Literal[
+    "int",
+    "float",
+    "str",
+    "bool",
+    "dict",
+    "list",
+    "object",
+    "datetime",
+    "date",
+]
```

Comment:
Some extras:

```suggestion
SchemaTypes = Literal[
    "int",             # Standard integer
    "float",           # Floating point number
    "str",             # String
    "bool",            # Boolean
    "dict",            # Dictionary
    "list",            # List
    "list[int]", "list[float]", "list[str]", "list[bool]", "list[dict]", "list[datetime]", "list[date]", "list[complex]", "list[list]", 
    "object",          # Generic object
    "datetime",        # Datetime object
    "date",            # Date object
    "complex",         # Complex number
    "ndarray",         # Numpy array
    "timedelta",       # Time delta object
    "bytes",           # Bytes
    "Decimal",         # Decimal numbers for high precision
    "Fraction",        # Fraction type
    'int8', 'int16', 'int32', 'int64',
        'uint8', 'uint16', 'uint32', 'uint64',
        'float16', 'float32', 'float64', 'float128',
        'complex64', 'complex128', 'complex256',
        'timedelta64', 'datetime64', 'timestamp',
]
```
otherwise how do we want to deal with nested structures properly? 

I'm almost inclined to say don't do any validation, but allow people to register their own, or by default register our one here that they can override and/ or extend.

Reply:
nvm -- I see this isn't validated against...